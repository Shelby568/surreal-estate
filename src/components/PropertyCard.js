import React from 'react';
import '../styles/PropertyCard.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faHome, faBath, faBed, faStar, faEnvelope,
} from '@fortawesome/free-solid-svg-icons';
import propTypes from 'prop-types';

const PropertyCard = ({
  _id,
  title,
  bedrooms,
  bathrooms,
  price,
  email,
  type,
  city,
  onSaveProperty,
  userID,
}) => (

  <div className="propertyCard">
      <div className="card">
        <div className="picture">
          <FontAwesomeIcon icon={faHome} size="4x" />
        </div>
        <div className="title" data-testid="title-id">{title}</div>
        <div className="type" data-testid="type-id">{type}</div>
        <div className="bathroom" data-testid="bathrooms-id">
          <FontAwesomeIcon icon={faBath} />
          {' '}
          {bathrooms}
        </div>
        <div className="bedroom" data-testid="bedrooms-id">
          <FontAwesomeIcon icon={faBed} />
          {' '}
          {bedrooms}
        </div>
        <div className="price" data-testid="price-id">
          Â£
          {price}
        </div>
        <div className="city" data-testid="city-id">{city}</div>
        <div className="email" data-testid="email-id">
          <button
            type="button"
            href={email}
            className="email-button"
          >
            <FontAwesomeIcon icon={faEnvelope} className="email" />
            {' '}
            Email
          </button>
        </div>
        {userID && (
        <button
          type="button"
          href="#"
          onClick={() => onSaveProperty(_id)}
          className="save-button"
        >
          <FontAwesomeIcon icon={faStar} />
          {' '}
          Save
        </button>
        )}
      </div>

  </div>
);

PropertyCard.propTypes = {
  _id: propTypes.string.isRequired,
  title: propTypes.string.isRequired,
  type: propTypes.string.isRequired,
  bedrooms: propTypes.string.isRequired,
  bathrooms: propTypes.string.isRequired,
  price: propTypes.number.isRequired,
  city: propTypes.string.isRequired,
  email: propTypes.string.isRequired,
  onSaveProperty: propTypes.func.isRequired,
  userID: propTypes.string.isRequired,
};

export default PropertyCard;
