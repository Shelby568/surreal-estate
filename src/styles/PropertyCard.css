.propertyCard {
   display: inline;
   text-align: center;
   padding: 15px; 
   border-color: #d3d3d3;
   border: solid 2px;
   border-radius: 6px;
   box-shadow: 3px 3px #d3d3d3;
   margin-top: 5px;
   max-width: 325px;
   max-height: 400px;
}

.propCard {
   display: flex;
   justify-content: space-between;
   flex-wrap: wrap;
   padding: 20px;
}

.picture, .bedroom, .bathroom {
    color: #373a47;
}

.email {
   color: white;
}

.email-button, .save-button {
   background-color: #373a47;
   color: white;
   font-weight: bold;
   height: 35px;
   width: 100px;
   border: solid;
   border-radius: 6px;
   cursor: pointer;
}

@media only screen and (max-width: 380px) {
   .propertyCard {
      margin-top: 50px;
      width: 75%;
      height: 75%;
   }
}