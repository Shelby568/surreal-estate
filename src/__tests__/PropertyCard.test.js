import React from 'react';
import { render } from '@testing-library/react';
import PropertyCard from '../components/PropertyCard';

describe('Properties', () => {
  const properties = [
    {
      _id: '123a',
      title: '2 Bed Townhouse',
      type: 'Semi-Detached',
      bathrooms: '1',
      bedrooms: '2',
      price: 80000,
      city: 'Manchester',
      email: 'email1@email.com',
    },
    {
      _id: '123b',
      title: 'Period Property',
      type: 'Detached',
      bathrooms: '2',
      bedrooms: '4',
      price: 150000,
      city: 'Liverpool',
      email: 'email2@email.com',
    },
  ];
  it('checks component renders correctly', () => {
    const { asFragment } = render(<PropertyCard properties={properties} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders to correct amount of component props', () => {
    const { getAllByTestId } = render(<PropertyCard properties={properties} />);

    expect(getAllByTestId('title-id')).toHaveLength(1);
    expect(getAllByTestId('type-id')).toHaveLength(1);
    expect(getAllByTestId('bathrooms-id')).toHaveLength(1);
    expect(getAllByTestId('bedrooms-id')).toHaveLength(1);
    expect(getAllByTestId('price-id')).toHaveLength(1);
    expect(getAllByTestId('city-id')).toHaveLength(1);
    expect(getAllByTestId('email-id')).toHaveLength(1);
});
});
