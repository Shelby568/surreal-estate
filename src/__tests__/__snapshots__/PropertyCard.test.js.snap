// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties checks component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="propertyCard"
  >
    <div
      class="card"
    >
      <div
        class="picture"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-home fa-w-18 fa-4x "
          data-icon="home"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="title"
        data-testid="title-id"
      />
      <div
        class="type"
        data-testid="type-id"
      />
      <div
        class="bathroom"
        data-testid="bathrooms-id"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bath fa-w-16 "
          data-icon="bath"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
            fill="currentColor"
          />
        </svg>
         
      </div>
      <div
        class="bedroom"
        data-testid="bedrooms-id"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bed fa-w-20 "
          data-icon="bed"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
            fill="currentColor"
          />
        </svg>
         
      </div>
      <div
        class="price"
        data-testid="price-id"
      >
        Â£
      </div>
      <div
        class="city"
        data-testid="city-id"
      />
      <div
        class="email"
        data-testid="email-id"
      >
        <button
          class="email-button"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope fa-w-16 email"
            data-icon="envelope"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
              fill="currentColor"
            />
          </svg>
           Email
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
